### DISCUSSION TITLE
Why can't I run the phases out of `nix develop`?

### DISCUSSION THREAD

**Discussion by Elias-Graf** [rust-lang/crates.io#9415] [Q&A]
I'm trying to debug a flake (since it doesn't build with `nix build`, but does when running `cargo build` inside the dev shell), by executing the exact phases. But the devShell environment seems to be different from the build environment. For example, none of the phases seem to be populated:

```
[elias@perseus:~/Git/cosmic-files]$ echo $phases
buildPhase

[elias@perseus:~/Git/cosmic-files]$ echo $buildPhase
{ echo "------------------------------------------------------------"; echo " WARNING: the existence of this path is not guaranteed."; echo " It is an internal implementation detail for pkgs.mkShell."; echo "------------------------------------------------------------"; echo; # Record all build inputs as runtime dependencies export; } >> "$out"
```

I'm unsure if this is a general nix concept I'm not understanding, but I'm pretty sure that I could execute `eval "$buildPhase"` when building a simple "Hello World C" flake.

My flake is pretty much the one from the simple template I think, but here is the shell part if it's of any use:

```nix
        devShells.default = craneLib.devShell {
          # Inherit inputs from checks.
          checks = self.checks.${system};

          # Additional dev-shell environment variables can be set directly
          # MY_CUSTOM_DEVELOPMENT_VAR = "something else";

          # Extra inputs can be added here; cargo and rustc are provided by default.
          packages = [
            # pkgs.ripgrep
          ];
        };
```

Help and pointers greatly appreciated.

---
In case anyone is interested, what isn't working when running `nix build`: Pretty sure the issue is that the translation files do not get copied to the build environment due to the `cleanCargoSource`. But I would like to confirm that before just trying, so I can learn something. Thus, I'm trying to get the phases to work, which possible might get me the `src` if I run unpack(?).

**Comment by Turbo87** [comment]
this appears to be unrelated to crates.io. might be better to ask for help on the nix project.

**Reply by Elias-Graf** [reply]
I'm very sorry, I have no idea how I managed to post this here instead of on https://github.com/ipetkov/crane. Thank you for informing me about my mistake!

**Comment by eth3lbert** [comment]
I could share the flake.nix I used if you're interested https://gist.github.com/eth3lbert/c93e70dc44f80d81d993b75cb91e05bc.

(Ah, just noticed the problem has nothing to do with crates.io.😅)

**Reply by Elias-Graf** [reply]
Thank you for your answer, though :) Help is always appreciated!
